pkg/dataplane/client_test.go

```diff
@@ -26,3 +38,176 @@ func TestNewClient(t *testing.T) {
 		t.Fatalf("Client is nil")
 	}
 }
+
+func TestGetUserAssignedMSI(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name        string
+		goMockCall  func(swaggerClient *mock.MockswaggerMSIClient)
+		request     UserAssignedMSIRequest
+		expectedErr error
+	}{
+		{
+			name:        "IdenityURL not specified",
+			goMockCall:  func(swaggerClient *mock.MockswaggerMSIClient) {},
+			request:     UserAssignedMSIRequest{ResourceID: validResourceID, TenantID: validTenantID},
+			expectedErr: errInvalidRequest,
+		},
+		{
+			name:        "IdenityURL not a URL",
+			goMockCall:  func(swaggerClient *mock.MockswaggerMSIClient) {},
+			request:     UserAssignedMSIRequest{IdentityURL: "bogus", ResourceID: validResourceID, TenantID: validTenantID},
+			expectedErr: errInvalidRequest,
+		},
+		{
+			name:        "ResourceID not specified",
```

Comment:
Should we add a test case / validation code for `resource ID is not of format Resource ID` or something similar to what we have for IdentityURL and TenantID? We've done similar work in the RP's static validation: 

- https://github.com/Azure/ARO-RP/blob/master/pkg/api/v20240812preview/openshiftcluster_validatestatic_test.go#L1209-L1219

I'm not sure if this would ever happen, but might be good to cover.